docs/configuration/identity-providers/oidc.md

```diff
@@ -283,6 +300,23 @@ required: yes
 
 The shared secret between Authelia and the application consuming this client. Currently this is stored in plain text.
 
+#### public
+<div markdown="1">
+type: bool
+{: .label .label-config .label-purple }
+default: false
+{: .label .label-config .label-blue }
+required: no
+{: .label .label-config .label-green }
+</div>
+
+The public option enables a client where the end user or a malicious application could reasonably obtain the [client id](#id)
+and [client secret](#secret). This is particularly useful for SPA's and CLI tools. This option requires (and allows)
+setting the [client secret](#secret) to a blank string.
+
+In addition to the standard redirect URI requirements, the host must be `localhost` or `127.0.0.1`. Alternatively a
+redirect URI of exactly `urn:ietf:wg:oauth:2.0:oob` is valid.
```

Comment:
Can you point to a documentation mentioning this URI? I found some links related to Google using that URI but nothing really described in a RFC.

Reply:
Basically I had a look at what Dex did in this area. However on closer inspection they allow any redirect URI for public clients now. I think forcing clients which redirect to `localhost`, `127.0.0.*`, or `urn:ietf:wg:oauth:2.0:oob` to be public is longeterm going to be a reasonable safeguard. However upon inspection we should ensure  the OOB URI is not used with the token response type: https://github.com/dexidp/dex/blob/master/server/oauth2.go#L557-L559

I think what we should do for now is:

- allow public clients which is required to support cli tools safely like argo
- prevent localhost redirect uri's on anything other than public clients
- not support the OOB urn

Thoughts @clems4ever ?

Reply:
So if fosite already does it, let's trust them imo. It's better that the RFC implementation remains fosite duty altogether.

Reply:
But we can keep some part of the documentation regarding the fact that only OOB should be used though.